CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.1)
+project(fastor VERSION 0.6.4)
+
+option(BUILD_TESTING "Build the testing tree." ON)
+
+set(FASTOR_INCLUDE_DIR "${fastor_SOURCE_DIR}")
+
+add_library(Fastor INTERFACE)
+
+target_include_directories(
+    Fastor INTERFACE
+        $<BUILD_INTERFACE:${FASTOR_INCLUDE_DIR}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+if(BUILD_TESTING)
+    enable_testing()
+
+    add_subdirectory(tests)
+endif()
+
+include(GNUInstallDirs)
+install(
+    DIRECTORY "${fastor_SOURCE_DIR}/Fastor"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+install(
+    TARGETS Fastor
+    EXPORT FastorTargets)
+
+export(
+    TARGETS Fastor
+    NAMESPACE Fastor::
+    FILE "${fastor_BINARY_DIR}/FastorTargets.cmake")
+export(PACKAGE Fastor)
+install(
+    EXPORT FastorTargets
+    NAMESPACE Fastor::
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Fastor")
+
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+    "${fastor_SOURCE_DIR}/data/FastorConfig.cmake.in"
+    "${fastor_BINARY_DIR}/FastorConfig.cmake"
+    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Fastor"
+    PATH_VARS CMAKE_INSTALL_INCLUDEDIR)
+write_basic_package_version_file(
+    "${fastor_BINARY_DIR}/FastorConfigVersion.cmake"
+    VERSION ${fastor_VERSION}
+    COMPATIBILITY SameMajorVersion
+    ARCH_INDEPENDENT)
+install(
+    FILES
+        "${fastor_BINARY_DIR}/FastorConfigVersion.cmake"
+        "${fastor_BINARY_DIR}/FastorConfig.cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Fastor")
+
+configure_file(
+    "${fastor_SOURCE_DIR}/data/fastor.pc.in"
+    "${fastor_BINARY_DIR}/fastor.pc" @ONLY)
+install(
+    FILES "${fastor_BINARY_DIR}/fastor.pc"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
```

Comment:
Not sure about the pkg-config id here. The project's names is `Fastor` but pkg-config convention seems to recommend lower case. Using `fastor` may be confusing though... Opinions welcome.